src/services/user.js

```diff
@@ -0,0 +1,22 @@
+import orgApi from "./";
+import { authHeader } from "utils/formatting";
+
+async function put(data) {
+  let response;
+  try {
+    const user = { ...data };
+    delete event['user'];
+    response = await orgApi.put("/user", data, {
+      headers: authHeader(data.user),
+    });
+    // TO-DO: If password changed, create and store new auth credentials
+    return response;
+  } catch (err) {
+    console.log("Event PUT error: ", err.response || err);
+    return err;
+  }
+}
+
+export default eventServices = {
+  put
+};
```

Comment:
I would recommend you delete the export and do this

```js
export async function put(data) {
}
```

so that you import like this:

```js
import { put } from 'services/user'
```

I'm pretty sure the syntax you're using there will cause a linting error, since you're assigning to a variable you don't use. And I also think it's a bit weird to use default exports in this way, where you're exporting an object with your function in it.

Philosophically, I think I'm starting to land in the camp of using default exports only for components, and normal exports for everything else. But I've even read convincing pieces about never using default exports (because with default exports, you can import something with a different name than how it's declared in the file, which makes it hard to refactor later).

Also, `update` would probably be a more evocative function name than `put`.

Reply:
I did the export object because we use the Axios services in React components, and using an object was easier, both to keep track of which services we were using (event vs. user vs. auth, etc), and to prevent the individual function imports at the top of a component from getting out of hand. (I saw this pattern in another project but for the life of me can't find it rn)
But because I did it that way to tidy up the component's imports, I found it cleaner to have all of the function names at the bottom of the services files so you can see which functions you can use at a glance.
I also used put since I was trying to mirror the Axios function names and corresponding HTTP requests as closely as I could.
Open to discussing all that and making changes if you think this pattern could be improved.